specs/driver_spec.rb

```diff
@@ -77,4 +99,93 @@
       driver.average_rating.must_equal 0
     end
   end
+
+  describe "Total Revenue" do
+    before do
+      start_time = Time.parse('2015-05-20T12:14:00+00:00')
+      end_time = start_time + 30
+
+      @driver = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV", vin: "1C9EVBRM0YBC564DZ")
+
+      trip_one = RideShare::Trip.new({id: 2, driver: @driver, passenger: nil, start_time: start_time, end_time: end_time, cost: 40, rating: 5})
+
+      trip_two = RideShare::Trip.new({id: 8, driver: @driver, passenger: nil, start_time: start_time, end_time: end_time, cost: 25, rating: 3})
+
+      trip_three = RideShare::Trip.new({id: 5, driver: @driver, passenger: nil, start_time: start_time, end_time: nil, cost: 0, rating: nil})
+
+      @driver.add_trip(trip_one)
+      @driver.add_trip(trip_two)
+      @driver.add_trip(trip_three)
+
+    end
+    it 'calculate total revenue from driver' do
+      @driver.total_revenue.must_be_kind_of Float
+      @driver.total_revenue.must_equal 49.36
+    end
+    it 'calculate the average revenue per hour from driver' do
+      @driver.average_revenue.must_be_kind_of Float
+      @driver.average_revenue.must_equal 24.68
+    end
```

Comment:
You do a good job of catching edge cases here, especially for in-progress trips. One other you might ask is "what happens if the driver has no trips".

However, I think these tests could be better organized. Standard practice is to have one big outer `describe` block for the whole class, and then a sub-`describe` for each method. If there is some setup that's common to multiple sub-`describe`s, you can put it in a `before` block in the top-level `describe`. Something like this:

```ruby
describe 'Driver' do
  before do
    @driver_no_trips = Driver.new(...)
    @driver_complete_trips = Driver.new(...)
    # ... add a couple completed trips to @driver_complete_trips
    @driver_incomplete_trips = Driver.new(...)
    # ... add a couple incomplete trips
  end
  describe 'total_revenue' do
    it 'sums up trip revenues'
    it 'returns 0 if there are no trips'
    it 'ignores in-progress trips'
  end
end
```